sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -1403,15 +1396,17 @@ void ProgramManager::addImages(pi_device_binaries DeviceBinary) {
     }
     // Otherwise assume that the image contains all kernels associated with the
     // module
-    KernelSetId &KSId = m_OSModuleKernelSets[M];
+    if (!m_OSModuleKernelSet)
```

Comment:
Minor: since the whole notion of having kernels be defined by name + OS module pair goes away and now a kernel name is a unique identifier regardless of the OS module the kernel is coming from, a name like "UniversalKernelSet" would probably be more appropriate.

As a note, we can probably get rid of the whole logic for this path completely. From what I remember, this was originally added to support cases where no kernel name info was available in the fat binary. Given that there have been several ABI breaking windows since then, now we should always expect this info to be present. This is definitely something outside the scope of this change though.